plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,317 @@
+    GridDataFactory.defaultsFor(artifactIdField).align(SWT.FILL, SWT.CENTER)
+        .applyTo(artifactIdField);
+    artifactIdField.addModifyListener(pageValidator);
+
+    Label versionLabel = new Label(mavenCoordinatesGroup, SWT.NONE);
+    versionLabel.setText("Version:"); //$NON-NLS-1$
+    versionField = new Text(mavenCoordinatesGroup, SWT.BORDER);
+    versionField.setText(defaultVersion);
+    GridDataFactory.defaultsFor(versionField).align(SWT.FILL, SWT.CENTER).applyTo(versionField);
+    versionField.addModifyListener(pageValidator);
+  }
+
+  /** Create UI for specifying App Engine project details */
+  private void createAppEngineProjectDetailsArea(Composite container) {
+    ModifyListener pageValidator = new PageValidator();
+
+    // Java package name
+    Label packageNameLabel = new Label(container, SWT.NONE);
+    packageNameLabel.setText("Java package:"); //$NON-NLS-1$
+    javaPackageField = new Text(container, SWT.BORDER);
+    GridData javaPackagePosition = new GridData(GridData.FILL_HORIZONTAL);
+    javaPackagePosition.horizontalSpan = 2;
+    javaPackageField.setLayoutData(javaPackagePosition);
+    javaPackageField.addModifyListener(pageValidator);
+
+    // App Engine Project ID
+    Label projectIdLabel = new Label(container, SWT.NONE);
+    projectIdLabel.setText("App Engine Project ID: (optional)"); //$NON-NLS-1$
+    projectIdField = new Text(container, SWT.BORDER);
+    GridData projectIdPosition = new GridData(GridData.FILL_HORIZONTAL);
+    projectIdPosition.horizontalSpan = 2;
+    projectIdField.setLayoutData(projectIdPosition);
+    projectIdField.addModifyListener(pageValidator);
+  }
+
+  protected void openLocationDialog() {
+    DirectoryDialog dialog = new DirectoryDialog(getShell());
+    dialog.setText("Please select the location to contain generated project");
+    String loc = dialog.open();
+    if (loc != null) {
+      locationField.setText(loc);
+      checkPageComplete();
+    }
+  }
+
+  protected void checkPageComplete() {
+    setPageComplete(validatePage());
+  }
+
+  /**
+   * Validate and report on the contents of this page
+   * 
+   * @return true if valid, false if there is a problem
+   */
+  public boolean validatePage() {
+    setMessage(null);
+    setErrorMessage(null);
+    
+    // order here should match order of the UI fields
+
+    String location = locationField.getText().trim();
+    if (!useDefaults() && location.isEmpty()) {
+      setMessage("Please provide a location", INFORMATION);
+      return false;
+    }
+
+    if (!validateMavenSettings()) {
+      return false;
+    }
+    if (!validateGeneratedProjectLocation()) {
+      return false;
+    }
+
+    String packageName = getPackageName();
+    if (!JavaPackageValidator.validate(packageName)) {
+      setErrorMessage(MessageFormat.format("Illegal Java package name: {0}.", packageName));
+      return false;
+    }
+
+    String projectId = getAppEngineProjectId();
```

Comment:
could refactor into a validateAppEngineProjectId method if you like to parallel above